feat: add cosmos plugin

anilcse opened the pull request:
<!-- Use this template by filling in information and copy and pasting relevant items out of the html comments. -->

# Relates to:
<!-- LINK TO ISSUE OR TICKET -->
NA

<!-- This risks section is to be filled out before final review and merge. -->

# Risks
<!--
Low, medium, large. List what kind of risks, and what could be effected.
-->
NA

# Background

## What does this PR do?
This PR adds a **Cosmos plugin** for Eliza, enabling agents to interact with any Cosmos SDK-based chains (e.g., Osmosis, Cosmos Hub, Akash).

## What kind of change is this?
<!--
Bug fixes (non-breaking change which fixes an issue)
Improvements (misc. changes to existing features)
Features (non-breaking change which adds functionality)
Updates (new versions of included code)
-->
**Features** (non-breaking change which adds functionality)

<!-- This "Why" section is most relevant if there is no linked issue explaining why. 
If there is a related issue it might make sense to skip this why section. -->
<!--
## Why are we doing this? Any context or related work?
-->

# Documentation changes needed?
<!--
My changes do not require a change to the project documentation.
My changes require a change to the project documentation.
If a docs change is needed: I have updated the documentation accordingly.
-->
My changes do not require a change to the project documentation.

# Testing

## Where should a reviewer start?
- Look over the `plugin-cosmos` folder structure, especially `providers/wallet.ts`, `actions/transfer.ts`, and any environment or index files.
- Check the integration tests in `tests/wallet.test.ts`.

## Detailed testing steps
1. **Install dependencies** for the plugin-cosmos package:
   ```bash
   pnpm install

proteanx commented:
This looks good from what I can tell so far. Might be good to go ahead and list ```COSMOS_RPC_URL``` in the .env.example so users know they can set their rpc if they choose.

I haven't tested this yet, but can soon.

anilcse commented:
> This looks good from what I can tell so far. Might be good to go ahead and list `COSMOS_RPC_URL` in the .env.example so users know they can set their rpc if they choose.

Sure, I will add it. This would definitely avoid extra calls to chain-registry if the user can simply provide all these config details, optionally

Freytes reviewed (approved):
follows the standard of Goat and other finanacal plugins.

bertux committed:
feat: add Arthera plugin from EVM plugin model

feat: rename plugin from EVM to Arthera and update README

feat: remove bridge and swap actions from Arthera plugin

feat: update transfer examples and templates to use AA instead of ETH

feat: update viem dependency to version 2.21.58 and adjust pnpm-lock.yaml

feat: remove unused LiFi dependencies and clean up type definitions in Arthera plugin

feat: remove bridge actions and templates from Arthera plugin

feat: remove swap actions and templates from Arthera plugin

feat: update EVM naming to Arthera

feat: update README and types for Arthera mainnet integration

feat: update plugin to use Arthera instead of mainnet

fix: add required devDependencies

fix: remove switchChain

fix: update _options type to Record<string, unknown> in transferAction

fix: correct log message format in transfer action to include wallet client address

test: enhance transfer tests with additional wallet provider and address validation

Plugin arthera merge (#3)

* feat: added arthera to default character and agent

* feat: renamed EVM_PRIVATE_KEY by ARTHERA_PRIVATE_KEY

* fix: roll back core package

* fix: workspace: version

---------

Co-authored-by: Arthera Node <node@arthera.bask.guru>

fix: run transfer test only if private key provided

fix: add missing newline at end of package.json and tsconfig.json files

marcellodesales committed:
:bug: generating package-lock.json for docusaurus updates

marcellodesales committed:
:whale: Just changing Dockerfile.docs with build comments

marcellodesales committed:
:scroll: update package-lock.json with current docs package.json

marcellodesales committed:
:scroll: update pnpm-lock.yaml with docusaurus changes

rawgroundbeef committed:
fix(solana token provider): await cached data

ai16z-demirix committed:
test: fixing generation.test.ts. Adding tests for memory, goals and provider

ai16z-demirix committed:
Merge branch 'develop' into test/goals-memory-provider

MacsDickinson committed:
Merge branch 'develop' into 1838--fix-faq-link-in-contributing-md

poodengDev committed:
Fix faq link on contributing.md

poodengDev committed:
Update contributing link aswell to faq

poodengDev committed:
fix broken links

shakkernerd committed:
Merge branch 'develop' into 1838-fix-faq-link-mtoor

tcm390 committed:
sort timelines based on the llm score

Gajesh2007 committed:
fix: fixed the anthropic regex

Gajesh2007 committed:
Merge branch 'main' of https://github.com/Gajesh2007/eliza

Gajesh2007 committed:
Merge main: Add reclaim adapter and resolve conflicts

Gajesh2007 committed:
fix: conflicts

Gajesh2007 committed:
Merge branch 'develop' into main

aisjdoaisjdoi2123 committed:
Update .env.example

lalalune committed:
Merge pull request #1835 from IterateFast-Labs/develop

Update README_KOR.md

lalalune committed:
Merge pull request #1832 from Valetek/refactor/readme

refactor: typo in readme

lalalune committed:
Merge pull request #1847 from manotoor/1838-fix-faq-link-mtoor

Fix faq link on contributing.md

lalalune committed:
Merge pull request #1787 from RobertSloan22/patch-1

Create README.md

lalalune committed:
Merge pull request #1827 from macfly-base/main

Update c3po.character.json

mcp97 committed:
Merge branch 'develop' into test/goals-memory-provider

aisjdoaisjdoi2123 committed:
Add files via upload

aisjdoaisjdoi2123 committed:
Add files via upload

aisjdoaisjdoi2123 committed:
Merge branch 'develop' into develop

lalalune committed:
Merge branch 'feat/genlayer-plugin' of https://github.com/yeagerai/eliza into HEAD

daniel-farina committed:
Add Stargaze plugin

daniel-farina committed:
Merge branch 'develop' into stargaze-merge-conflicts

lalalune committed:
Merge branch 'bugfix/broken-ci-docs-missed-frozen' of https://github.com/marcellodesales/elizaOS into HEAD

lalalune committed:
Merge branch 'patch-4' of https://github.com/affaan-m/eliza into HEAD

lalalune committed:
Merge branch '1838--fix-faq-link-in-contributing-md' of https://github.com/MacsDickinson/eliza into HEAD

lalalune committed:
Merge branch 'develop' of https://github.com/inferanetwork/eliza into HEAD

lalalune committed:
Merge branch 'main' of https://github.com/robin-rrt/eliza into HEAD

daniel-farina committed:
Merge branch 'develop' into stargaze-merge-conflicts

lalalune committed:
Merge branch 'develop' of https://github.com/blockydevs/eliza into HEAD

lalalune committed:
Merge branch 'roshan/plugin-tee-marlin' of https://github.com/marlinprotocol/eliza into HEAD

lalalune committed:
fix lock

odilitime committed:
Add DAO donation ask & dev discord

odilitime committed:
Merge pull request #1867 from elizaOS/odilitime-patch-1

docs: Add DAO donation ask & dev discord

tcm390 committed:
clean code and add comment

tcm390 committed:
add MAX_ACTIONS_PROCESSING

tcm390 committed:
Merge branch 'main' into tcm-max-actions

tcm390 committed:
remove unused variable

tcm390 committed:
remove unused variable

tcm390 committed:
add EmbeddingModelSettings type

tcm390 committed:
get the info from model settings instead of hardcoced number

tcm390 committed:
remove unused variable

tcm390 committed:
fix provider path

tcm390 committed:
Merge branch 'main' into tcm-refactor-modelConfig

tcm390 committed:
Merge branch 'develop' into tcm-refactor-modelConfig

odilitime committed:
Merge branch 'develop' into docs/readme-ro

odilitime committed:
fix bad merge

odilitime committed:
revert 1834 changes

odilitime committed:
audit fix

shakkernerd committed:
Merge pull request #1872 from elizaOS/fix-postinstall-script

fix: Fix postinstall script

shakkernerd committed:
Merge branch 'develop' into tcm-refactor-modelConfig

odilitime committed:
Merge branch 'develop' into docs/readme-ro

odilitime committed:
Merge pull request #1770 from pedronovikovborges/docs/readme-ro

docs: add Romanian README translation

tcm390 committed:
resolve conflict

shakkernerd committed:
Merge branch 'develop' into test/goals-memory-provider

odilitime committed:
Merge branch 'develop' into fix/twitter-profile-cache-1638

shakkernerd committed:
Merge pull request #1840 from ai16z-demirix/test/goals-memory-provider

test: adding more tests for goals,memory and provider. Fixing generation.test.ts

odilitime committed:
Merge branch 'develop' into fix/twitter-profile-cache-1638

odilitime committed:
Merge pull request #1782 from augchan42/fix/twitter-profile-cache-1638

fix: remove twitter profile caching (#1638)

shakkernerd committed:
Merge branch 'develop' into add-devcontainer

shakkernerd committed:
Merge pull request #1807 from jazzvaz/add-devcontainer

feat: Add devcontainer

shakkernerd committed:
Merge branch 'develop' into fix/await-cached-prices-2

shakkernerd committed:
Merge pull request #1828 from bentatum/fix/await-cached-prices-2

fix(solana token provider): await cached data

odilitime committed:
support more debians distros

odilitime committed:
Merge branch 'develop' into fix-postinstall-script

shakkernerd committed:
Merge pull request #1875 from odilitime/fix-postinstall-script

chore: support more debians distros

tcm390 committed:
Merge branch 'develop' into tcm-refactor-modelConfig

shakkernerd committed:
Merge pull request #1805 from elizaOS/tcm-refactor-modelConfig

feat: refactor model config

rawgroundbeef committed:
chore: enable strict null checks

ai16z-demirix committed:
test: adding test setup for telegram client

ai16z-demirix committed:
test: adding test setup for telegram client - improving message manager tests

ai16z-demirix committed:
Merge branch 'develop' into test/client-telegram

YoungPhlo committed:
docs: Add "What Did You Get Done This Week? 8" notes

kylebuildsstuff committed:
Add OpenWeather plugin

Arrowana committed:
Use recommended settings in jupiter swap

JoeyKhd committed:
upd: fixed broken API Documentation URL

JoeyKhd committed:
upd: fixed path

Gajesh2007 committed:
Merge branch 'elizaOS:main' into main

bertux committed:
chore: bump version to 0.1.8-alpha.1 for plugin-arthera

Gajesh2007 committed:
Merge old into develop with regenerated pnpm-lock.yaml

Gajesh2007 committed:
Merge develop into main

Gajesh2007 committed:
fix: settings error

fabianhug committed:
Merge pull request #1558 from Gajesh2007/main

feat: added new plugin - zktls - reclaim

augchan42 committed:
Merge branch 'elizaOS:main' into fix/twitter-wildcard-targets-1556

HashWarlock committed:
feat: add remote attestation action

augchan42 committed:
just add needed change to run unit test

shakkernerd committed:
chore: pnpm lock file

YoungPhlo committed:
docs: Re-fix timestamp links in WDYGDTW 4 notes

shakkernerd committed:
Merge branch 'develop' into pr-1884

shakkernerd committed:
chore: fix broken pnpm lock file

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
fix: change file extension from rs -> ts

shakkernerd committed:
Merge branch 'develop' into pr-1885

YoungPhlo committed:
docs: Fix timestamp links for Dev School 3

shakkernerd committed:
Merge branch 'develop' into chore/jupiter-swap-recommended-settings

shakkernerd committed:
Merge pull request #1882 from Arrowana/chore/jupiter-swap-recommended-settings

feat: Use recommended settings in jupiter swap

shakkernerd committed:
Merge branch 'develop' into fix-broken-docs-api-url

shakkernerd committed:
Merge pull request #1881 from JoeyKhd/fix-broken-docs-api-url

Fixed broken API Documentation URL

shakkernerd committed:
Merge branch 'develop' into pr-1880

shakkernerd committed:
Merge branch 'develop' into test/client-telegram

shakkernerd committed:
Merge pull request #1879 from ai16z-demirix/test/client-telegram

test: adding test setup for telegram client

YoungPhlo committed:
docs: Fix timestamp links for Dev School 4

shakkernerd committed:
Merge branch 'develop' into pr-1878

YoungPhlo committed:
docs: Re-fix timestamp links in WDYGDTW 5 notes

shakkernerd committed:
fix: conflicting Readme.md files

zhourunlai committed:
fix case sensitive path

shakkernerd committed:
chore: explicitly set node version and bump pnpm version

shakkernerd committed:
Merge branch 'develop' into fix_case_sensitive_path

shakkernerd committed:
Merge pull request #1892 from zhourunlai/fix_case_sensitive_path

fix: remove case sensitive path

shakkernerd committed:
Merge branch 'develop' into fix/tests

shakkernerd committed:
chore: remove pnpm install with no frozen lockfile flag

shakkernerd committed:
chore: pnpm lock file

odilitime committed:
hyperfi endpoint, clean up multer/imports/console logs, notes

shakkernerd committed:
Merge pull request #1896 from odilitime/hfi

feat: hyperfy support

shakkernerd committed:
chore: make viem a global package & remove from plugins package.json

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
chore: add no frozen install flag

odilitime committed:
Merge branch 'develop' into docs/community-streams

shakkernerd committed:
chore: remove flag and disable removing lock file

shakkernerd committed:
chore: changes to packages

shakkernerd committed:
chore: add frozen flag

zhourunlai committed:
fix twitter undefined properties

tomguluson92 committed:
update goplus to the safety module

tomguluson92 committed:
Update README.md

RatakondalaArun committed:
fix: add callback to the evaluators for client-telegram

Commit:
Add Dutch (Nederlands) translation

0xbeekeeper committed:
Update index.ts

lalalune committed:
Merge branch 'develop' into add-dutch-translation

lalalune committed:
Merge pull request #1909 from xzrfax/add-dutch-translation

Add Dutch (Nederlands) translation

lalalune committed:
Merge pull request #1905 from zhourunlai/fix_twitter_undefined_properties

fix: twitter result obj undefined properties

shakkernerd committed:
chore: remove conflict package

shakkernerd committed:
Merge branch 'develop' into fix/tests

shakkernerd committed:
Revert "chore: enable strict null checks"

This reverts commit fcf913797d7345b3e418c817d88347b5ae87868c.

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #1893 from elizaOS/fix/tests

chore: fix integrations and smoke tests

shakkernerd committed:
Merge branch 'develop' into 1906--fix-evaluate-callback-for-client-telegram

bodhi-crypo committed:
Update summarize_conversation.ts

JoeyKhd committed:
upd: resolved a bunch of type related errors and ensure project compiles properly

JoeyKhd committed:
Merge branch 'develop' into fix-types-and-build

Lukapetro committed:
chore: resolve merge conflicts with develop

rawgroundbeef committed:
fix(core): check evaluators for null

shakkernerd committed:
Merge pull request #1908 from RatakondalaArun/1906--fix-evaluate-callback-for-client-telegram

fix: add callback to the evaluators for client-telegram

shakkernerd committed:
Merge branch 'develop' into fix-types-and-build

shakkernerd committed:
Merge pull request #1916 from bodhi-crypo/main

chore(client-discord): fix SUMMARIZE prompt

shakkernerd committed:
Merge branch 'develop' into fix/runtime-evaluate-null

shakkernerd committed:
Merge pull request #1918 from bentatum/fix/runtime-evaluate-null

fix(core): check evaluators for null

shakkernerd committed:
Merge branch 'develop' into fix-types-and-build

shakkernerd committed:
Merge pull request #1917 from JoeyKhd/fix-types-and-build

fix: resolved a bunch of type related errors and ensure project compiles

shakkernerd committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge pull request #1907 from tomguluson92/main

feats: diagram update

tcm390 committed:
add followinf option for timeline action

bertux committed:
Merge branch 'develop' into plugin-arthera-alpha

zhourunlai committed:
fix empty input text

tcm390 committed:
Merge branch 'develop' into tcm-max-actions

shakkernerd committed:
Merge pull request #1824 from elizaOS/tcm-max-actions

feat: Optimize Agent Action Processing by Prioritizing Timelines and Limiting Actions Per Cycle

wtfsayo committed:
Merge branch 'develop' into feature/add-binance-plugin

petryshkaCODE committed:
Fix typo Update CHANGELOG.md

wtfsayo committed:
Merge pull request #1812 from Lukapetro/feature/add-binance-plugin

Feature/add binance plugin

wtfsayo committed:
Merge branch 'develop' into Fix-typo

wtfsayo committed:
Merge pull request #1922 from petryshkaCODE/Fix-typo

Fix typo Update CHANGELOG.md

wtfsayo committed:
Merge branch 'develop' into fix_empty_input_text

wtfsayo committed:
Merge branch 'develop' into goplus-security

wtfsayo committed:
Merge pull request #1898 from 0xbeekeeper/goplus-security

feat: add GoPlus Security Plugin to enhance security for agent

wtfsayo committed:
Merge branch 'develop' into plugin-arthera-alpha

JoeyKhd committed:
upd: hide character settings secrets from api route

mcp97 committed:
Merge branch 'develop' into fix_empty_input_text

shakkernerd committed:
Merge branch 'develop' into hide-secrets

shakkernerd committed:
Merge pull request #1923 from JoeyKhd/hide-secrets

API route update

mcp97 committed:
Merge branch 'develop' into fix_empty_input_text

ai16z-demirix committed:
test: adding tests for discord-client

jonathangus committed:
Merge branch 'develop' into dynamic-template

bertux committed:
Merge branch 'develop' into plugin-arthera-alpha

shakkernerd committed:
Merge pull request #1919 from zhourunlai/fix_empty_input_text

fix: handle empty input text to avoid memory content empty error

shakkernerd committed:
Merge branch 'develop' into tests/client-discord

shakkernerd committed:
Merge pull request #1927 from ai16z-demirix/tests/client-discord

test: adding tests for discord-client

shakkernerd committed:
Merge branch 'develop' into docs/community-streams

shakkernerd committed:
Merge pull request #1894 from YoungPhlo/docs/community-streams

docs: Add "What Did You Get Done This Week? #8" notes

ai16z-demirix committed:
test: adding tests for twitter-client

proteanx committed:
rm unused variable x_server_url

tcm390 committed:
Limit the number of timelines fetched

tcm390 committed:
make a constant

tcm390 committed:
add comment

sinecose committed:
Merge branch 'develop' into zkevm-no-warning

shakkernerd committed:
Merge pull request #1931 from elizaOS/tcm-limit-fetchingTimelines

fix: Limit the number of timelines fetched

shakkernerd committed:
Merge branch 'develop' into rmxserver

shakkernerd committed:
Merge pull request #1930 from proteanx/rmxserver

fix: rm unused variable X_SERVER_URL

HashWarlock committed:
Merge branch 'develop' into dynamic-template

HashWarlock committed:
Merge pull request #1467 from jonathangus/dynamic-template

feat: Make templates in composeContext dynamic

threewebcode committed:
docs: it is processAction instead of triggerAction

There is only processAction in runtime instance. Update them correctly.

shakkernerd committed:
Merge pull request #1937 from threewebcode/develop

docs: it is processAction instead of triggerAction

Commit:
feat: support for eternalai provider can make request with chain_id extra data in body

odilitime committed:
Merge branch 'develop' of https://github.com/elizaos/eliza into feat/sol-agent-kit

odilitime committed:
Merge pull request #1373 from renlulu/feat/sol-agent-kit

feat: solana token deploy using solana agent kit

odilitime commented on packages/plugin-cosmos/package.json:
```diff
@@ -0,0 +1,30 @@
+        "bignumber.js": "9.1.2",
+        "chain-registry": "*",
+        "cosmjs-utils": "*",
+        "node-cache": "5.1.2",
```
do we want another caching system? we should unify this with our caching in the future

Commit:
Merge branch 'develop' into eternal-ai-pr-chainid

odilitime commented:
@anilcse can you fix the conflicts and we'll get this merged in. Doesn't look like we have the ability to do that for you on this one.

bendanzhentan committed:
refactor: remove unused imports

shakkernerd committed:
Merge branch 'develop' into chore/remove-unused-imports

shakkernerd committed:
Merge pull request #1941 from bendanzhentan/chore/remove-unused-imports

chore(plugin-conflux): remove unused imports

shlokkhemani committed:
chore: add embedding tests

ag-wnl committed:
Merge branch 'develop' into zkevm-no-warning

ag-wnl committed:
Merge pull request #1744 from sinecose/zkevm-no-warning

feat(plugin-cronoszkevm): rm not used imports

0xRider committed:
chore: more specific rpc urls

0xRider committed:
Merge branch 'develop' into chore/more-specific-rpc-urls

darwintree committed:
feat: improve twitter paragraph splitting if containing url

shakkernerd committed:
Merge pull request #1938 from TrustlessComputer/eternal-ai-pr-chainid

feat: support for eternalai provider can make request with chain_id extra data in body

shakkernerd committed:
Merge branch 'develop' into chore/more-specific-rpc-urls

Commit:
feat: support for eternalai provider can write request/response, change name in env

shakkernerd committed:
Merge pull request #1945 from 0xRider/chore/more-specific-rpc-urls

chore: more specific rpc urls

Commit:
Merge branch 'develop' into eternal-ai-pr-chainid

shakkernerd committed:
Merge pull request #1948 from TrustlessComputer/eternal-ai-pr-chainid

feat: support for eternalai provider can write request/response log info

shakkernerd committed:
Merge branch 'develop' into add-embedding-tests

shakkernerd committed:
Merge pull request #1944 from shlokkhemani/add-embedding-tests

chore: add embedding tests

darwintree committed:
Merge branch 'develop' into twitter-url

JoeyKhd committed:
upd: fixed issue with c3po character json

wtfsayo committed:
Merge branch 'develop' into plugin-arthera-alpha

JoeyKhd committed:
upd: updated validateCharacterConfig to give better feedback

wtfsayo committed:
Merge pull request #1818 from bertux/plugin-arthera-alpha

feat: new plugin Arthera Chain

lalalune committed:
Merge pull request #1951 from JoeyKhd/fix-c3po

fix: fixed error in C3PO and improved error handling feedback

wtfsayo committed:
Merge branch 'develop' into pr/1810

kesar committed:
added fraxtal

kesar committed:
added fraxtal

kesar committed:
Merge branch 'develop' into patch-1

wtfsayo committed:
update .env.example and gitignore

wtfsayo committed:
Merge pull request #1810 from KONFeature/1809-add-pglite-adapter

feat: add pglite db adapter

kesar committed:
Merge branch 'develop' into patch-1

wtfsayo committed:
Merge branch 'develop' into pr/1808

ai16z-demirix committed:
Merge branch 'develop' into test/client-twitter

ag-wnl committed:
Merge pull request #1954 from kesar/patch-1

feat: Add fraxtal chain to evm

wtfsayo committed:
Update pnpm-lock.yaml

wtfsayo committed:
Merge branch 'develop' into coinPricePlugin

wtfsayo committed:
Merge pull request #1808 from proteanx/coinPricePlugin

feat: add coin price plugin (CoinMarketCap, CoinGecko & CoinCap)

shakkernerd committed:
Merge branch 'develop' into test/client-twitter

cygaar committed:
fix: abstract plugin readme

wtfsayo committed:
Merge pull request #1963 from cygaar/abstract_readme

fix: abstract readme

shakkernerd committed:
Merge branch 'develop' into test/client-twitter

tcm390 committed:
Merge branch 'develop' into twitter-url

shakkernerd committed:
Merge pull request #1947 from darwintree/twitter-url

feat: improve twitter paragraph splitting if containing url

shakkernerd committed:
Merge branch 'develop' into test/client-twitter

shakkernerd committed:
Merge pull request #1959 from ai16z-demirix/test/client-twitter

test: adding tests for twitter-client

anilcse committed:
fix conflicts

anilcse committed:
Merge develop

anilcse committed:
fix merge conflicts

anilcse commented:
Looks like #1826 got merged before this. So closing it.
